pandas/core/generic.py

```diff
@@ -629,11 +669,88 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
         Returns
         -------
         renamed : type of caller
+
+        See Also
+        --------
+        pandas.NDFrame.rename
+        pandas.Index.rename
+
+        Examples
+        --------
+        >>> df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
+        >>> df.rename_axis("foo")  # scalar, alters df.index.name
+             A  B
+        foo
+        0    1  4
+        1    2  5
+        2    3  6
+        >>> df.rename_axis(lambda x: 2 * x)  # function: alters labels
+           A  B
+        0  1  4
+        2  2  5
+        4  3  6
+        >>> df.rename_axis({"A": "ehh", "C": "see"}, axis="columns")  # mapping
+           ehh  B
+        0    1  4
+        1    2  5
+        2    3  6
+        """
+        is_scalar_or_list = (
+            (not com.is_sequence(mapper) and not callable(mapper)) or
+            (com.is_list_like(mapper) and not isinstance(mapper,
```

Comment:
I find `is_list_like` & co ambiguous, because I don't know if `list` means to me what it means to the person who wrote it.  the function name should describe the properties being tested for (e.g. `is_immuatable_sequence` rather than `is_tuple_like`).  #12056 is properly the place to discuss that.